src/lqcut.jl

```diff
@@ -1,58 +1,77 @@
+     DEP_edgesigma::Float64 = 3.0 , mode::Symbol = :gaussian, lower_exclusion::Float64 = 0.005, upper_exclusion::Float64 = 0.98, drift_cutoff_sigma::Float64 = 2.0,
+     e_expression::Union{String,Symbol}="e")
 
-    #Using fixed values for DEP, can be changed to use values from DEP fit from Energy calibration 
-    lq_DEP_dt = lq_norm[DEP_left .< e_cal .< DEP_right]
+
+    #calculate DEP edges
+    DEP_left = DEP_µ - DEP_edgesigma * DEP_σ
+    DEP_right = DEP_µ + DEP_edgesigma * DEP_σ
+
+    #cut data to DEP peak
+    lq_DEP = lq_norm[DEP_left .< e_cal .< DEP_right]
     t_tcal = ustrip.(tdrift[DEP_left .< e_cal .< DEP_right])
 
     #lq cutoff
-    #sort array to exclude outliers
-    sort_lq = sort(lq_DEP_dt)
-    low_cut = Int(round(length(sort_lq) * lower_exclusion)) # cut lower 0.5%
-    high_cut = Int(round(length(sort_lq) * upper_exclusion)) # cut upper 2%
-    lq_prehist = fit(Histogram, lq_DEP_dt, range(sort_lq[low_cut], sort_lq[high_cut], length=100))
+    #sort array to exclude outliers for better fit
+    low_cut_value = quantile(lq_DEP, lower_exclusion)  # 0.5% quantile
+    high_cut_value = quantile(lq_DEP, upper_exclusion)  # 2% quantile
+
+    ideal_bin_width = LegendSpecFits.get_friedman_diaconis_bin_width(lq_DEP[low_cut_value .< lq_DEP .< high_cut_value])
+    ideal_length = Int(round(abs(high_cut_value - low_cut_value) / ideal_bin_width))
+
+    lq_prehist = fit(Histogram, lq_DEP, range(low_cut_value, high_cut_value, length=ideal_length))
     lq_prestats = estimate_single_peak_stats(lq_prehist)
+
     lq_start = lq_prestats.peak_pos - 3*lq_prestats.peak_sigma
     lq_stop = lq_prestats.peak_pos + 3*lq_prestats.peak_sigma
 
-    lq_edges = range(lq_start, stop=lq_stop, length=51)
-    lq_hist_DEP = fit(Histogram, lq_DEP_dt, lq_edges)
-
-    lq_DEP_stats = estimate_single_peak_stats(lq_hist_DEP)
-    lq_result, lq_report = LegendSpecFits.fit_binned_gauss(lq_hist_DEP, lq_DEP_stats)
+    lq_result, lq_report = fit_binned_trunc_gauss(lq_prehist, (low=lq_start, high=lq_stop, max=NaN))
     µ_lq = mvalue(lq_result.μ)
     σ_lq = mvalue(lq_result.σ)
 
     #set cutoff in lq dimension for later fit
-    lq_lower = µ_lq - drift_cutoff_sgima * σ_lq 
-    lq_upper = µ_lq + drift_cutoff_sgima * σ_lq 
+    lq_lower = µ_lq - drift_cutoff_sigma * σ_lq 
+    lq_upper = µ_lq + drift_cutoff_sigma * σ_lq 
 
     #t_tcal cutoff; method dependant on detector type
-    if detectortype in [:ppc, :bege, :coax] 
-        drift_prehist = fit(Histogram, t_tcal, range(minimum(t_tcal), stop=maximum(t_tcal), length=100))
+    if mode == :gaussian 
+        
+        #ideal bin width for histogram
+        ideal_bin_width = LegendSpecFits.get_friedman_diaconis_bin_width(t_tcal)
+        ideal_length = Int(round((maximum(t_tcal) - minimum(t_tcal)) / ideal_bin_width))
+
+        drift_prehist = fit(Histogram, t_tcal, range(minimum(t_tcal), stop=maximum(t_tcal), length=ideal_length))
```

Comment:
Again:
* remove `LegendSpecFits` (not needed)
* do we need `length = ideal_length` or could we use `step = ideal_bin_width` here?